### PR TITLE
Support for goal TSRs

### PR DESCRIPTION
**Pull request by jeking04**
This implements goalset TSRs. Trajectory wide constraints and start sets not yet supported.

### CODE REVIEW

File: src/TSRRobot.cpp [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+            }
+
+            // Check for axis flip, marked by Bw values being backwards
+            bool bFlipAxis = false;
+            if(Bw(j,0) > Bw(j,1)){
+                Bw(j,0) = -Bw(j,0);
+                Bw(j,1) = -Bw(j,1);
+
+                bFlipAxis = true;
+            }
```
Comment (jeking04) [inline src/TSRRobot.cpp]: This is a convention set copied from CBiRRT. I believe the desired functionality is for you to specify something like:

Bw(0,0)= -5, Bw(0,1) = -10

When what you actual want is:
Bw(0,0) = 5, Bw(0,1) = 10

but along an axis pointing in the opposite direction of convention. So you flip the sign, and then set a flag which is later used to flip the axis direction.

File: include/or_ompl/RobotStateSpace.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+         */
+        std::vector<double> getValues() const;
+
+        /**
+         * @return An ordered list of indices this state corresponds to
+         */
+        std::vector<int> getIndices() const { return _indices; }
+
+    private:
+        std::vector<int> _indices;
```
Comment (jeking04) [inline include/or_ompl/RobotStateSpace.h]: States are created via a call to allocState on the state space. The state space doesn't directly keep a handle to states its allocated. So I think technically there is no guarentee of this.  In practice, most objects in ompl (including the base planner class and the PathGeometric objects) take a SpaceInformationPtr, which has internally a shared_ptr to the StateSpace. So it is almost always true that the StateSpace would outlive the useful life of the State. Thoughts?

File: src/OMPLPlanner.cpp [hunk 3]
```diff
@@ -128,21 +126,36 @@ bool OMPLPlanner::InitPlan(OpenRAVE::RobotBasePtr robot,
...
+				RAVELOG_ERROR("Goal configuration is in collision.\n");
+				return false;
+			}
+
+			ScopedState q_goal(m_state_space);
+			for (size_t i = 0; i < num_dof; i++) {
+				q_goal->values[i] = m_parameters->vgoalconfig[i];
+			}
+			m_simple_setup->setGoalState(q_goal);
+		}
```
Comment (jeking04) [inline src/OMPLPlanner.cpp]: Fixed. Now I throw an exception if there are goal_chains and a goal config supplied.

File: src/TSRRobot.cpp [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+ 
+}
+
+bool TSRRobot::construct() {
+
+    if(_initialized){
+        RAVELOG_DEBUG("[TSRRobot] Already initialized. Skipping construct.");
+        return _initialized;
+    }
```
Comment (jeking04) [inline src/TSRRobot.cpp]: Changed. Now an exception is thrown.

**Comment by mkoval** [comment]
LGTM. Thanks for handling the merge with `master`!

File: src/TSRRobot.cpp (hunk 1 above)
Comment (mkoval) [inline src/TSRRobot.cpp]: I can't think of a legitimate reason to use the former. Can we remove this? Or print a warning if we hit this case?

**Pull request merged**